src/poc/miner_poc_statem.erl

```diff
@@ -360,6 +345,31 @@ handle_event(_EventType, _EventContent, Data) ->
     lager:warning("ignoring event [~p] ~p", [_EventType, _EventContent]),
     {keep_state, Data}.
 
+
+handle_mining(BlockHash, PinnedLedger, #data{blockchain = _Chain, address = Challenger,
+                                             mining_timeout = MiningTimeout, secret = Secret} = Data)->
+    case find_request(BlockHash, Data) of
+        {ok, Block} ->
+            Height = blockchain_block:height(Block),
+            %% TODO discuss making this delay verifiable so you can't punish a hotspot by
+            %% intentionally fast-challenging them before they have the block
+            %% Is this timer necessary ??
+            timer:sleep(?BLOCK_PROPOGATION_TIME),
+            %% save the block hash in which we found the poc request, if we crash out in targeting we can pick back up
+            %% this will be nulled if we revert back to requesting or go forward to receiving
+            save_data(Data#data{targeting_data={targeting, BlockHash, PinnedLedger}}),
```

Comment:
Shouldn't we pass the Data with the updated targeting_data through to handle_targeting?

Reply:
you are right, thats an oversight.  Will fix